### PR TITLE
Feature/bdi donation creation when no contact first name opportunity name is created with null

### PR DESCRIPTION
**Pull request by mdeleon0**
# Critical Changes

# Changes
- When creating Opportunities, Batch Data Import will now exclude blank Contact First Names from Opportunity names. Previously, blank Contact first names were displaying in the Opportunity name as "null."

# Issues Closed

# New Metadata

# Deleted Metadata

### CODE REVIEW

File: src/classes/BDI_Donations.cls [hunk 1]
```diff
@@ -433,25 +437,29 @@ public class BDI_Donations {
         }
 
         // now copy over all the data to the Opp
-        for (string strDIField : mapDIFieldToOppField.keySet()) {
+        for (String dataImportField : mapDIFieldToOppField.keySet()) {
+
+            String donationRecordTypeName = UTIL_Namespace.StrTokenNSPrefix('Donation_Record_Type_Name__c').toLowercase();
```
Comment (lmeerkatz) [inline src/classes/BDI_Donations.cls]: We should be able to move these variables outside of the for loop, right?

Comment (mdeleon0) [reply src/classes/BDI_Donations.cls]: you are right.

**Comment by lmeerkatz** [comment]
**lurch:attach W-029130

**Comment by LurchTheButler** [comment]
Tracking <a href="https://foundation.my.salesforce.com/a2x80000007fOhyAAE" target="_blank">W-029130</a>

**Comment by cpose** [comment]
QA Approval.

**Pull request merged**